src/api/app/jobs/project_do_project_copy_job.rb

```diff
@@ -1,7 +1,41 @@
 class ProjectDoProjectCopyJob < ApplicationJob
   queue_as :quick
 
+  attr_accessor :project, :params
   def perform(project_id, params)
-    Project.find(project_id).do_project_copy(params)
+    self.project = Project.find(project_id)
+    self.params = params
+    do_project_copy
+  end
+
+  private
+
+  def do_project_copy
+    User.find(params[:user]).run_as do
+      # copy entire project in the backend
+      begin
+        path = "/source/#{CGI.escape(project.name)}"
```

Comment:
what do you think about using a helper to build the url? `File.join` for example would be an option, no?

Reply:
`File.join` is about the file system and this is not windows vs unix - these are URLs. next to that: this code is in around 1000 places within obs.

The backend lib uses placeholders like `http_post(['/source/:project', project_name]`, but this API is not exported and putting this call into the backend library is *way* outside of the scope of this PR.

Reply:
then please use `URL.join` if you prefer https://apidock.com/ruby/URI/join/class

Reply:
OBS uses in 0 places `URL.join`. What makes this place special?

Reply:
> OBS uses in 0 places `URL.join`. What makes this place special?

It's not really an argument, the only question that matters is if it's a better pattern or not.